[[enrichments]>worker22] ruby - Ruby exception occurred: no implicit conversion of nil into String

## Describe the bug
```
[ERROR] 2022-11-26 07:54:05.540 [[enrichments]>worker14] ruby - Ruby exception occurred: no implicit conversion of nil into String {:class=>"TypeError", :backtrace=>["(ruby filter code):68:in `block in filter_method'", "org/jruby/RubyArray.java:1865:in `each'", "(ruby filter code):67:in `block in filter_method'", "/usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-filter-ruby-3.1.8/lib/logstash/filters/ruby.rb:96:in `inline_script'", "/usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-filter-ruby-3.1.8/lib/logstash/filters/ruby.rb:89:in `filter'", "/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:159:in `do_filter'", "/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:178:in `block in multi_filter'", "org/jruby/RubyArray.java:1865:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:175:in `multi_filter'", "org/logstash/config/ir/compiler/AbstractFilterDelegatorExt.java:134:in `multi_filter'", "/usr/share/logstash/logstash-core/lib/logstash/java_pipeline.rb:301:in `block in start_workers'"]}
```

## X-Reference issues
_(Cross reference any user stories that this bug might be affecting)_

## Steps To Reproduce
start the enrichment pipeline. I'm using logstash 8.5.2

## Expected behavior
no error should  be seen

## Additional context
The following components in enrichment make use of ruby filer, but i don't understand what is the culprit

```
./02_ecs_data_type.conf
./04_timestamp.conf
./11_related_hosts.conf
./12_related_user.conf
./13_related_ip.conf
./14_related_hash.conf
./16_related_mac.conf
./93_mitre.conf
./94_remove_empty_n_truncate.conf
```

--- a/config/enrichments/94_remove_empty_n_truncate.conf
+++ b/config/enrichments/94_remove_empty_n_truncate.conf
@@ -63,17 +63,19 @@ filter {
         invalid_value_list = ["-", "null", "nil", "n/a", "\'\'"]
 
         def walk_hash(parent, path, hash)
+          paths = []
           path << parent if parent
           hash.each do |key, value|
-            walk_hash(key, path, value) if value.is_a?(Hash)
-            @paths << (path + [key]).map {|p| "[" + p + "]" }.join("")
+            paths += walk_hash(key, path, value) if value.is_a?(Hash)
+            computed_path = (path + [key]).map {|p| "[" + p.to_s + "]" }.join("")
+            paths << computed_path
           end
           path.pop
+          return paths
         end
 
-        @paths = []
-        walk_hash(nil, [], event.to_hash)
-        @paths.each do |k|
+        paths = walk_hash(nil, [], event.to_hash)
+        paths.each do |k|
           v = event.get(k)
           if v.nil? || (v.respond_to?(:empty?) && v.empty?)
             event.remove(k)
@@ -89,7 +91,7 @@ filter {
                   next
                 else
                   event.set(k, v.to_s[0, 1023])
-		end
+		            end
               end
             else
               # check if its an array

